benchmark: add --expose_internals switch

bzoz opened the pull request:
##### Checklist
- [X] `make -j4 test` (UNIX), or `vcbuild test nosign` (Windows) passes
- [X] commit message follows commit guidelines

##### Affected core subsystem(s)
benchmark


##### Description of change
<!-- Provide a description of the change below this comment. -->

Adds `--expose_insternals` switch to benchmark runner. This makes `misc/freelist.js` and any other benchmark requiring access to internal modules run properly on Windows.

cc @nodejs/benchmarking

gibfahn commented:
Nit: `--expose_insternals` -> `--expose_internals` in the commit message.

AndreasMadsen commented on benchmark/run.js:
```diff
@@ -36,7 +36,8 @@ if (format === 'csv') {
   const filename = benchmarks[i];
-  const child = fork(path.resolve(__dirname, filename), cli.optional.set);
+  const child = fork(path.resolve(__dirname, filename), cli.optional.set,
+                     { execArgv: ['--expose_internals'] });
```
This should extend `process.execArgv` not overwrite it.

bzoz replied:
We don't need it, but it will be helpful for someone who tries to run this one benchmark

AndreasMadsen replied:
`common.js` ensures that benchmarks are always forked, so there is no way to execute a single benchmark without `--expose_internals` being automatically applied.

bzoz replied:
It is for people that try `node benchmark\misc\freelist.js` and encounter problems.

AndreasMadsen replied:
Don't you think it would be better to move the `require('internal/freelist')` inside main, instead of having all these `--expose_internals` around.

bzoz replied:
Thats a great idea!

AndreasMadsen commented:
I think you should add this to `benchmark/common.js` not `benchmark/run.js`. Otherwise you will have to add it too all files that forks the benchmarks (`compare.js`, `scatter.js`, `run.js`).

bzoz commented:
Updated, PTAL

Unfortunately, we need to add this everywhere - in `compare.js`, `scatter.js` and `run.js`.

AndreasMadsen commented:
> Unfortunately, we need to add this everywhere - in compare.js, scatter.js and run.js.

Please elaborate on why you can't add it here: https://github.com/nodejs/node/blob/master/benchmark/common.js#L139L141

bzoz commented:
When `run.js` runs benchmarker file it will fail on `require('internals/..')` before reaching `common.createBenchmark`. It will never reach that line in `common.js`.

bzoz commented:
Updated again. @AndreasMadsen you had great idea. Moved the `require` to `main`, now only `common.js` needs the `--expose_internals` and everything works.

Fishrock123 commented on benchmark/misc/freelist.js:
```diff
@@ -1,13 +1,13 @@
 });
 
 function main(conf) {
+  const FreeList = require('internal/freelist').FreeList;
```
What's up with this? Maybe add a comment?

bzoz replied:
`require` was moved there, so it will work with adding `-expose_internals` only in `common.js`. It was the result of discusion here: https://github.com/nodejs/node/pull/8547#pullrequestreview-202699

bzoz commented:
Updated, PTAL

bzoz commented:
If there are no further objections I would like to land this tomorrow.

bzoz commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/4285/

bzoz commented:
Landed in https://github.com/nodejs/node/commit/99a2dd03cd286176d639ec898a08f27c1567f85c. @AndreasMadsen - again, thanks for great input.

AndreasMadsen commented:
Awesome work 👍
